### PR TITLE
Normative: Clarify validity of negative expanded year 0.

### PR DESCRIPTION
**Pull request by EricIO**
Following a discussion in
https://github.com/tc39/proposal-temporal/issues/1753 there is some
reason to clarify the parsing of a negative expanded year 0 (-000000).

There is currently some discrepancy in how different implementors
parse this, some rejecting and some accepting a negative expanded year
0.

### CODE REVIEW

**Comment by ljharb** [comment]
Can you provide more details about current web reality here?

A normative change requires consensus, which includes the implementers who’d need to change being willing to do so, so we need to be clear about who those are and what needs changing.

**Comment by arhadthedev** [comment]
@ljharb, here are the results for `eshost -se 'Date.parse("-000000-01-01T00:00:00.000Z")'`:

```bash
#### ChakraCore, engine262, V8
NaN

#### GraalJS, JavaScriptCore, Moddable XS, QuickJS, SpiderMonkey
-62167219200000
```

Details on engines for reproducibility:

```
┌────────────────┬───────────┬────────────────────────────────────────────┬──────┬────────────────────────────────────────────────┐
│ name           │ type      │ path                                       │ args │ tags                                           │
├────────────────┼───────────┼────────────────────────────────────────────┼──────┼────────────────────────────────────────────────┤
│ ChakraCore     │ ch        │ /home/runner/.esvu/bin/chakra              │      │ 1_11_24,web                                    │
├────────────────┼───────────┼────────────────────────────────────────────┼──────┼────────────────────────────────────────────────┤
│ GraalJS        │ graaljs   │ /home/runner/.esvu/bin/graaljs             │      │ 21.2.0                                         │
├────────────────┼───────────┼────────────────────────────────────────────┼──────┼────────────────────────────────────────────────┤
│ JavaScriptCore │ jsc       │ /home/runner/.esvu/bin/jsc                 │      │ 283881,web                                     │
├────────────────┼───────────┼────────────────────────────────────────────┼──────┼────────────────────────────────────────────────┤
│ Moddable XS    │ xs        │ /home/runner/.esvu/bin/xs                  │      │ 10.5.0,embedded                                │
├────────────────┼───────────┼────────────────────────────────────────────┼──────┼────────────────────────────────────────────────┤
│ QuickJS        │ qjs       │ /home/runner/.esvu/bin/quickjs-run-test262 │      │ 2021-03-27,embedded                            │
├────────────────┼───────────┼────────────────────────────────────────────┼──────┼────────────────────────────────────────────────┤
│ SpiderMonkey   │ jsshell   │ /home/runner/.esvu/bin/sm                  │      │ 95.0a1#20211010214947,web                      │
├────────────────┼───────────┼────────────────────────────────────────────┼──────┼────────────────────────────────────────────────┤
│ V8             │ d8        │ /home/runner/.esvu/bin/v8                  │      │ 9.7.3,web                                      │
├────────────────┼───────────┼────────────────────────────────────────────┼──────┼────────────────────────────────────────────────┤
│ engine262      │ engine262 │ /home/runner/.esvu/bin/engine262           │      │ 0.0.1-f47d7da0653c7b2f0417395794a5259c2287cc44 │
└────────────────┴───────────┴────────────────────────────────────────────┴──────┴────────────────────────────────────────────────┘
```

> It’d be great to run https://npmjs.com/eshost 

Unfortunately, a newcomer will install a library and will stuck, so I'd rather refer to https://npmjs.com/eshost-cli (plus https://npmjs.com/esvu to properly install engines and feed them to eshost).

**Comment by ljharb** [comment]
Thanks, I've added this to the next meeting's agenda.

**Comment by EricIO** [comment]
Sorry for the silence. Here is an updated version that clarifies the text in the expanded year section to disallow -000000.

**Review by gibson042 — APPROVED** [review]

File: spec.html
Comment (michaelficarra) [inline spec.html:32070]:
```suggestion
          <p>Covering the full time value range of approximately 273,790 years forward or backward from 1 January 1970 (<emu-xref href="#sec-time-values-and-time-range"></emu-xref>) requires representing years before 0 or after 9999. ISO 8601 permits expansion of the year representation, but only by mutual agreement of the partners in information interchange. In the simplified ECMAScript format, such an expanded year representation shall have 6 digits and is always prefixed with a + or - sign. The year 0 is considered positive and must be prefixed with a + sign. The representation of the year 0 as -000000 is invalid. Strings matching the <emu-xref href="#sec-date-time-string-format">Date Time String Format</emu-xref> with expanded years representing instants in time outside the range of a time value are treated as unrecognizable by <emu-xref href="#sec-date.parse">`Date.parse`</emu-xref> and cause that function to return *NaN* without falling back to implementation-specific behaviour or heuristics.</p>
```

Two sentences. Also clarified.

Comment (EricIO) [reply spec.html:32070]: Github won't allow me to commit this suggestion. Perhaps someone with write permission must do it? Seems like a better wording to me so I'd be happy with it.

Comment (jmdyck) [reply spec.html:32070]: > Github won't allow me to commit this suggestion.

If you mean via the "Commit suggestion" button, yeah that's never worked.

Comment (ljharb) [reply spec.html:32070]: I believe it doesn't work when the suggestion is left by someone with write access. You can add it on the command line tho.

**Comment by ljharb** [comment]
@ptomato ping for getting this on your radar; is someone working on test262 tests for this change, so we can land this normative PR?

**Comment by ptomato** [comment]
Test262 tests for the Temporal part of it have already landed: https://github.com/tc39/test262/pull/3388/commits/4596c7edb3894de3ac853a5e3ab0f678b98d0a52

I assumed someone else was going to do the legacy Date part of it... it should be more straightforward than for Temporal, since IIUC only two entry points have to be covered. I can do it eventually, but I have to concentrate on Temporal stuff until the March plenary.

**Comment by ljharb** [comment]
I believe this PR can't land until those tests are done; it's fine if it needs to wait until after March, it'll just miss the ES2022 train.

**Comment by ptomato** [comment]
This now has tests at https://github.com/tc39/test262/pull/3470

**Review by bakkot — APPROVED** [review]

**Review by michaelficarra — APPROVED** [review]

**Comment by ljharb** [comment]
@EricIO would you mind signing the [IPR form](https://tc39.es/agreements/contributor/)? Thanks!

**Comment by EricIO** [comment]
> @EricIO would you mind signing the [IPR form](https://tc39.es/agreements/contributor/)? Thanks!

@ljharb Done!

**Pull request merged**
